service/signer/client/gnfd_sign_client.go

```diff
@@ -212,19 +218,32 @@ func (client *GreenfieldChainSignClient) DiscontinueBucket(ctx context.Context,
 		log.CtxErrorw(ctx, "failed to get private key", "err", err)
 		return nil, merrors.ErrSignMsg
 	}
-	log.Infow("succeed loaded km", "scope", scope)
+
+	client.mu.Lock()
+	defer client.mu.Unlock()
+	nonce := client.gcAccNonce + 1
 
 	msgDiscontinueBucket := storagetypes.NewMsgDiscontinueBucket(km.GetAddr(),
 		discontinueBucket.BucketName, discontinueBucket.Reason)
 	mode := tx.BroadcastMode_BROADCAST_MODE_SYNC
 	txOpt := &ctypes.TxOption{
 		Mode:     &mode,
 		GasLimit: client.gasLimit,
+		Nonce:    nonce,
 	}
 
 	resp, err := client.greenfieldClients[scope].BroadcastTx(ctx, []sdk.Msg{msgDiscontinueBucket}, txOpt)
 	if err != nil {
 		log.CtxErrorw(ctx, "failed to broadcast tx", "err", err, "discontinue_bucket", msgDiscontinueBucket.String())
+		if strings.Contains(err.Error(), "account sequence mismatch") {
+			// if nonce mismatch, reset nonce by querying the nonce on chain
+			nonce, err := client.greenfieldClients[scope].GetNonce()
+			if err != nil {
+				log.CtxErrorw(ctx, "failed to get gc account nonce", "err", err)
+				return nil, merrors.ErrDiscontinueBucketOnChain
+			}
+			client.gcAccNonce = nonce - 1
+		}
```

Comment:
Print this error log to simplify troubleshooting.

Reply:
Yes, the error had been printed in line 237.